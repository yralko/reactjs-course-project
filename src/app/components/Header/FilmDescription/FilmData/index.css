.FilmData {
  width: 70%;
  padding: 0 40px;
}
