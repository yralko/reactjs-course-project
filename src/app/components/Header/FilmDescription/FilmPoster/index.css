.FilmPoster {
  width: 30%;
  margin: 0 20px 20px 0;

}

img {
  width: 100%;
}
