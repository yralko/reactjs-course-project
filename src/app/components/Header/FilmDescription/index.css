.FilmDescription {
  display: flex;
}

.returnToSearch {
  position: absolute;
  right: 3%;
  top: 30px;
  padding: 5px 15px;
  text-transform: uppercase;
  font-weight: bold;
  border: 1px solid transparent;
  background-color: #ffffff;
  color: #f65260;
  transition: all .3s;
}

.returnToSearch:hover {
  background-color: #f65260;
  color: #ffffff;
  border-color: #ffffff;
}
