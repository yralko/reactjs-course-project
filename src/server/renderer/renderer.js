import React from 'react';
import { renderToString } from 'react-dom/server';
import { Provider } from 'react-redux';
import { StaticRouter as Router } from 'react-router-dom';
import App from '../../app/App';


export default (req, store, context) => {
  const root = (
    <Provider store={store}>
      <Router location={req.url} context={context}>
        <App />
      </Router>
    </Provider>
  );

  const content = renderToString(root); 

  return `
  <!doctype html>
  <html lang="en">
  <head>
    <meta charset="utf-8">
    <title>The HTML5 Herald</title>
    <base href="/" />
    <link rel="stylesheet" href="style.css" />
  </head>
  <body>
    <div id="root">${content}</div>
    <script>window.INITIAL_STATE = ${JSON.stringify(store.getState())}</script>
    <script src="bundle.js"></script>
  </body>
  </html>
  `;
};
