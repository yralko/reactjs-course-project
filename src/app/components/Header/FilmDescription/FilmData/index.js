import FilmData from './FilmData';

export default FilmData;
