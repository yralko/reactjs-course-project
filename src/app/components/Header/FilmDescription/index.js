import FilmDescription from './FilmDescription';

export default FilmDescription;
