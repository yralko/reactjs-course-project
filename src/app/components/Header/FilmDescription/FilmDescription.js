import React, { Component } from 'react';
import { connect } from 'react-redux';
import FilmData from './FilmData';
import FilmPoster from './FilmPoster';
import classes from './index.css';
import * as actions from '../../../store/actions';

const FilmDescription = (props) => {
  let renderedContent;

  if (props.fetchedFilms) {
    const currentFilm = props.fetchedFilms
      .filter(val => val.id === props.selectedFilmId)[0];

    renderedContent = (
      <div className={classes.FilmDescription}>
        <FilmPoster
          src={currentFilm.poster_path}
        />
        <FilmData
          filmData={{
            title: currentFilm.title,
            release_date: currentFilm.release_date,
            duration: currentFilm.runtime,
            overview: currentFilm.overview,
            rating: currentFilm.vote_average,
            genres: currentFilm.genres,
          }}
        />
      </div>
    );
  } else {
    renderedContent = <p>Loading ...</p>;
  }

  return (
    <div>
      <button
        className={classes.returnToSearch}
        onClick={() => props.returnToSearchbox()}
      >
        Search
      </button>
      {renderedContent}
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    selectedFilmId: state.selectedFilmId,
    fetchedFilms: state.fetchedFilms,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    returnToSearchbox: () => dispatch(actions.returnToSearchbox()),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(FilmDescription);
