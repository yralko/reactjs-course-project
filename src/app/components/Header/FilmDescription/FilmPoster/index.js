import FilmPoster from './FilmPoster';

export default FilmPoster;
