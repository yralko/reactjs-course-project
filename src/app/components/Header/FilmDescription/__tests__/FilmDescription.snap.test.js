import React from 'react';
import renderer from 'react-test-renderer';
import { FilmDescription } from '../FilmDescription';

test('matches snapshot', () => {
  const tree = renderer.create(<FilmDescription />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
