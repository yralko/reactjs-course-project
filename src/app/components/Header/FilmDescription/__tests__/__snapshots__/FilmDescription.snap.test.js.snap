// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <button
    className={undefined}
    onClick={[Function]}
  >
    Search
  </button>
  <p>
    Loading...
  </p>
</div>
`;
